### PR TITLE
feat(dashboard): AIOps log rate analysis, pattern analysis, and change point chart panel types

### LINKED ISSUE #4005: [Feature] Support Kibana dashboard AIOps panel types (log_rate_analysis, pattern_analysis, change_point_chart)
**Is your feature request related to a problem? Please describe.**

The `elasticstack_kibana_dashboard` resource cannot embed any of the three Kibana AIOps panel types — `aiops_log_rate_analysis`, `aiops_pattern_analysis`, and `aiops_change_point_chart`. Users assembling analytical dashboards that surface AIOps insights have to fall back to raw panel-level `config_json` blobs, losing typed validation and drift-safe planning.

All three panels have clean, flat, data-view-anchored API models, and follow the same panelkit passthrough conventions already established for other typed panels on this resource.

**Describe the API surface.**

All three panels share the standard panel wrapper (`type`, `grid`, optional `id`, `config`) plus the standard panelkit passthroughs (`title`, `description`, `hide_border`, `hide_title`, `time_range`). Their branch-specific fields:

**`aiops_log_rate_analysis`** — `KibanaHTTPAPIsAiopsLogRateAnalysis`

- `data_view_id` (required): the data view whose log rate is analysed.

**`aiops_pattern_analysis`** — `KibanaHTTPAPIsAiopsPatternAnalysis`

- `data_view_id` (required): source data view.
- `field_name` (required): text field on which to run pattern analysis.
- `minimum_time_range` (optional enum): `no_minimum` | `1_week` | `1_month` | `3_months` | `6_months`.
- `random_sampler_mode` (optional enum): `off` | `on_automatic` | `on_manual`.
- `random_sampler_probability` (optional float, range `[0.00001, 0.5]`): random-sampling probability, only meaningful when `random_sampler_mode = on_manual`.

**`aiops_change_point_chart`** — `KibanaHTTPAPIsAiopsChangePointChart`

- `data_view_id` (required): source data view.
- `metric_field` (required): field used by the aggregation function.
- `aggregation_function` (optional enum): `avg` | `max` | `min` | `sum`.
- `split_field` (optional string): field used to split change-point results.
- `partitions` (optional list of strings): specific split-field values to include (filter set).
- `max_series_to_plot` (optional float): maximum number of change points to visualise. Kibana default is 6.
- `view_type` (optional enum): `charts` | `table`.

None of the three panels defines drilldowns in the API model.

**Describe the solution you'd like.**

Three new typed panel blocks (`aiops_log_rate_analysis_config`, `aiops_pattern_analysis_config`, `aiops_change_point_chart_config`), following the panelkit conventions established by existing flat-config panels such as `sloburnrate` and `syntheticsmonitors`.

```hcl
panels = [
  {
    type       = "aiops_log_rate_analysis"
    panel_grid = { x = 0, y = 0, w = 24, h = 15 }
    aiops_log_rate_analysis_config = {
      data_view_id = "logs-view"
      title        = "Log rate spike detection"
      hide_title   = false
      hide_border  = false
      time_range = {
        from = "now-24h"
        to   = "now"
      }
    }
  },
  {
    type       = "aiops_pattern_analysis"
    panel_grid = { x = 0, y = 15, w = 24, h = 15 }
    aiops_pattern_analysis_config = {
      data_view_id               = "logs-view"
      field_name                 = "message"
      minimum_time_range         = "1_week"
      random_sampler_mode        = "on_manual"
      random_sampler_probability = 0.01
      title                      = "Log pattern analysis"
    }
  },
  {
    type       = "aiops_change_point_chart"
    panel_grid = { x = 0, y = 30, w = 24, h = 15 }
    aiops_change_point_chart_config = {
      data_view_id         = "metrics-view"
      metric_field         = "system.cpu.total.pct"
      aggregation_function = "avg"
      split_field          = "host.name"
      partitions           = ["host-a", "host-b", "host-c"]
      max_series_to_plot   = 6
      view_type            = "charts"
      title                = "CPU change points"
    }
  }
]
```

Each config block should reuse the existing panel handler registry (`internal/kibana/dashboard/registry.go`) and the panelkit primitives providing sibling-block mutual exclusion, ID plan modifiers, and the shared `hide_title` / `hide_border` / `title` / `description` / `time_range` passthroughs.

**Describe alternatives you've considered.**

- Raw `config_json`. Works but sacrifices validation and drift readability, as with all other typed panel types.
- Shipping the three AIOps panels as separate issues / separate PRs. Considered — rejected because the panels share structural conventions and shipping them as one coherent unit reads better in the review, in docs, and in release notes.

**Additional context.**

Relevant source paths on `main`:

- `generated/kbapi/kibana.gen.go`:
  - `KibanaHTTPAPIsKbnDashboardPanelTypeAiopsLogRateAnalysis` (line ~46685); embeddable `KibanaHTTPAPIsAiopsLogRateAnalysis` (line ~39193).
  - `KibanaHTTPAPIsKbnDashboardPanelTypeAiopsPatternAnalysis` (line ~46701); embeddable `KibanaHTTPAPIsAiopsPatternAnalysis` (line ~39206).
  - `KibanaHTTPAPIsKbnDashboardPanelTypeAiopsChangePointChart` (line ~46669); embeddable `KibanaHTTPAPIsAiopsChangePointChart` (line ~39157).
- `internal/kibana/dashboard/schema.go` — dashboard resource schema; `panelHandlers` registration lives in `internal/kibana/dashboard/registry.go`.
- Sibling flat-config handler for the panelkit pattern: `internal/kibana/dashboard/panel/sloburnrate/`, `internal/kibana/dashboard/panel/syntheticsmonitors/`.

Out of scope for this issue:

- Panel-level drilldowns on any of the three AIOps panels — the API models do not expose them.
- Any changes to Kibana's AIOps job or ML infrastructure resources.

### PR DESCRIPTION
**Pull request by tobio**
## Summary

Adds three typed panel config blocks to the `elasticstack_kibana_dashboard` resource, following the established panelkit handler pattern:

- `aiops_log_rate_analysis_config` (REQ-047) — anchored to a `data_view_id`, with the standard presentation passthroughs.
- `aiops_pattern_analysis_config` (REQ-048) — adds `field_name`, optional `minimum_time_range` / `random_sampler_mode` enums, and `random_sampler_probability` (plan-time validated to `[0.00001, 0.5]`).
- `aiops_change_point_chart_config` (REQ-049) — adds `metric_field`, optional `aggregation_function` / `view_type` enums, `split_field`, `partitions` (a `Set` to prevent order drift), and `max_series_to_plot`.

Each panel ships in its own package under `internal/kibana/dashboard/panel/aiops*/` with `schema.go`, `model.go`, `api.go`, plus unit and acceptance tests. All three apply REQ-009 null-preservation, reject simultaneous `config_json`, and use the `schema.Float32Attribute` pattern for float32 API fields to avoid refresh drift (matching the `time_slider` panel).

Closes the AIOps panels gap surfaced in #4005, where users previously had to fall back to raw `config_json` blobs.

Fixes #4005.

## Changelog

Customer impact: enhancement
Summary: Added `aiops_log_rate_analysis_config`, `aiops_pattern_analysis_config`, and `aiops_change_point_chart_config` typed panel blocks to `elasticstack_kibana_dashboard`, enabling AIOps panels with typed validation and drift-safe planning instead of raw `config_json`.

## Changes

- `internal/kibana/dashboard/panel/aiopslograteanalysis/` — new package
- `internal/kibana/dashboard/panel/aiopspatternanalysis/` — new package
- `internal/kibana/dashboard/panel/aiopschangepointchart/` — new package
- `internal/kibana/dashboard/panelkit/time_range.go` — shared panel time-range mapping helper
- `internal/kibana/dashboard/panelkit/nullpreserve.go` — added `PreserveFloat32`
- `internal/kibana/dashboard/models/aiops.go` — new config models
- `internal/kibana/dashboard/models/panel.go` — three new `PanelModel` fields
- `internal/kibana/dashboard/registry.go` — three handlers registered
- `docs/resources/kibana_dashboard.md` — regenerated

## Tests

- Unit tests per panel: `BuildConfig` round-trip (required-only + all-optional), `PopulateFromAPI` null-preservation, import, `config_json` rejection, handler round-trip.
- Acceptance tests per panel: required-only create, all-optional create, import, plan stability, and plan-time error tests (invalid enum/probability/config_json/panel-type).
- Shared multi-panel acceptance test verifying sibling-block mutual exclusion across all three AIOps panel types.

Local validation: `make build` (incl. golangci-lint), `go vet`, `go test ./internal/kibana/dashboard/...`, and `make check-openspec` all pass.

## OpenSpec

Change: `dashboard-aiops-panels` — all 26 tasks complete; delta specs add REQ-047, REQ-048, REQ-049.

### CODE REVIEW

File: internal/kibana/dashboard/panel/aiopslograteanalysis/api_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+// state stay null after read even when the API returns server-side values.
+func TestPopulateFromAPI_nullPreservation(t *testing.T) {
+	t.Parallel()
+
+	api := kbapi.KibanaHTTPAPIsAiopsLogRateAnalysis{
+		DataViewId:  "logs-*",
+		Title:       new("Server default title"),
+		Description: new("Server default description"),
+		HideTitle:   new(true),
+		HideBorder:  new(false),
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopslograteanalysis/api_test.go:126]: The `new(value)` form is valid Go 1.26.1 syntax (the `//go:fix inline` directive marks it as a gofmt modernizer rewrite of the traditional `&value` address-of pattern). The repo targets `go 1.26.1` (see go.mod), and CI confirms these test files compile and the unit tests pass (e.g. `TestBuildConfig_requiredOnly`, `TestPopulateFromAPI_import`). No change needed.

File: internal/kibana/dashboard/panel/aiopslograteanalysis/api_test.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+}
+
+// TestPopulateFromAPI_import verifies import (prior == nil) populates required fields and optional
+// fields only when the API returns non-nil values.
+func TestPopulateFromAPI_import(t *testing.T) {
+	t.Parallel()
+
+	api := kbapi.KibanaHTTPAPIsAiopsLogRateAnalysis{
+		DataViewId: "logs-*",
+		Title:      new("Imported title"),
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopslograteanalysis/api_test.go:172]: The `new(value)` form is valid Go 1.26.1 syntax (the `//go:fix inline` directive marks it as a gofmt modernizer rewrite of the traditional `&value` address-of pattern). The repo targets `go 1.26.1` (see go.mod), and CI confirms these test files compile and the unit tests pass (e.g. `TestBuildConfig_requiredOnly`, `TestPopulateFromAPI_import`). No change needed.

File: internal/kibana/dashboard/panel/aiopspatternanalysis/api_test.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+	t.Parallel()
+
+	mtr := kbapi.KibanaHTTPAPIsAiopsPatternAnalysisMinimumTimeRange("1_week")
+	rsm := kbapi.KibanaHTTPAPIsAiopsPatternAnalysisRandomSamplerMode("on_automatic")
+	api := kbapi.KibanaHTTPAPIsAiopsPatternAnalysis{
+		DataViewId:               "logs-*",
+		FieldName:                "message",
+		MinimumTimeRange:         &mtr,
+		RandomSamplerMode:        &rsm,
+		RandomSamplerProbability: new(float32(0.02)),
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopspatternanalysis/api_test.go:173]: The `new(value)` form is valid Go 1.26.1 syntax (the `//go:fix inline` directive marks it as a gofmt modernizer rewrite of the traditional `&value` address-of pattern). The repo targets `go 1.26.1` (see go.mod), and CI confirms these test files compile and the unit tests pass (e.g. `TestBuildConfig_requiredOnly`, `TestPopulateFromAPI_import`). No change needed.

File: internal/kibana/dashboard/panel/aiopschangepointchart/api_test.go [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+	agg := kbapi.KibanaHTTPAPIsAiopsChangePointChartAggregationFunction("avg")
+	vt := kbapi.KibanaHTTPAPIsAiopsChangePointChartViewType("charts")
+	api := kbapi.KibanaHTTPAPIsAiopsChangePointChart{
+		DataViewId:          "metrics-*",
+		MetricField:         "system.cpu.total.pct",
+		AggregationFunction: &agg,
+		SplitField:          new("host.name"),
+		Partitions:          &[]string{"host-a", "host-b"},
+		MaxSeriesToPlot:     new(float32(6)),
+		ViewType:            &vt,
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopschangepointchart/api_test.go:190]: The `new(value)` form is valid Go 1.26.1 syntax (the `//go:fix inline` directive marks it as a gofmt modernizer rewrite of the traditional `&value` address-of pattern). The repo targets `go 1.26.1` (see go.mod), and CI confirms these test files compile and the unit tests pass (e.g. `TestBuildConfig_requiredOnly`, `TestPopulateFromAPI_import`). No change needed.

File: internal/kibana/dashboard/panel/aiopspatternanalysis/api.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		},
+		"Failed to create AIOps pattern analysis panel",
+	)
+}
+
+// ValidatePanelConfig enforces presence of data_view_id and field_name for aiops_pattern_analysis panels.
+func (Handler) ValidatePanelConfig(_ context.Context, attrs map[string]attr.Value, attrPath path.Path) diag.Diagnostics {
+	return panelkit.ValidateDataViewFieldName(attrs, panelType+"_config",
+		"Invalid AIOps pattern analysis configuration", attrPath)
+}
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopspatternanalysis/api.go:73]: Addressed in d3ed0801a: the  block is now `Required: true` when `type = "aiops_pattern_analysis"` (matching the ML anomaly panel convention), so a bare panel without its config block now fails at plan time via the `RequiredIf` validator rather than producing an empty API payload.

File: internal/kibana/dashboard/panel/aiopschangepointchart/model.go [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+	if typeutils.IsKnown(existing.ViewType) {
+		existing.ViewType = changePointViewTypeValue(api.ViewType)
+	}
+	existing.MaxSeriesToPlot = panelkit.PreserveFloat32(existing.MaxSeriesToPlot, api.MaxSeriesToPlot)
+
+	// Partitions set: null-preserve. When the practitioner omitted it (null/unknown in state), keep null
+	// regardless of API-returned values; otherwise refresh from the API set (order-insensitive).
+	if typeutils.IsKnown(existing.Partitions) {
+		existing.Partitions = changePointPartitionsFromAPI(api.Partitions)
+	}
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopschangepointchart/model.go:143]: This is an intentional, documented design decision (see design.md Risks/Trade-offs and the spec'\''s "omitted → null" semantics for `partitions`). `partitions` is modelled as a filter set; an explicitly-empty set collapsing to null on read is the documented edge case and matches the established filter-set semantics across the resource. The unit test `TestPopulateFromAPI_partitionsOrderInsensitive` covers the order-insensitive round-trip.

File: internal/kibana/dashboard/panel/aiopspatternanalysis/api.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/panel/aiopspatternanalysis/api.go:73]: Addressed in d3ed0801a: the aiops_pattern_analysis_config block is now `Required: true` when `type = "aiops_pattern_analysis"` (matching the ML anomaly panel convention), so a bare panel without its config block now fails at plan time via the RequiredIf validator rather than producing an empty API payload.
